plugin/src/test/java/com/microsoft/java/bs/gradle/plugin/GradleBuildServerPluginTest.java

```diff
@@ -73,10 +75,9 @@ void testModelBuilder() throws IOException {
             dependency -> dependency.getModule().contains("gradle-api")
         ));
 
-        DefaultJavaExtension javaExtension = Conversions.toJavaExtension(
-            gradleSourceSet.getExtensions().get(SupportedLanguages.JAVA));
+        JavaExtension javaExtension =
```

Comment:
What about this: `JavaExtension javaExtension = (JavaExtension) gradleSourceSet.getExtensions()
.get(SupportedLanguages.JAVA.getBspName());`

I'm thinking that by writing like this, we can limit the convert interface only exist in `LanguageExtension`, which maybe makes it simpler?

Reply:
@jdneo Are you reviewing since the new commit - Github seems to think so but I can't tell for sure?
I've deleted the `Conversions` class and moved all this into the `LanguageExtension`.
I can squash the commits if you want

Reply:
I think I'm reviewing the latest one.

Because I see both `LanguageExtension` and `SupportedLanguage` has declares `convert()` method.

Reply:
Oh I see.  The reason I've put it in `SupportedLanguage` as well is that then you never need to cast it.
So no one can mismatch the types and names e.g.
`JavaExtension javaExtension = (JavaExtension) gradleSourceSet.getExtensions() .get(SupportedLanguages.SCALA.getBspName());`
or
`ScalaExtension javaExtension = (ScalaExtension) gradleSourceSet.getExtensions() .get(SupportedLanguages.JAVA.getBspName());`
Otherwise there's a lot of typecasting in the tests.  This way it's impossible to miscast.

Reply:
I see. What about giving it another name, maybe something like `getExtension()` or `getLanguageExtension()`? And give some document explaining the purpose of this method?

I'm trying to avoid introduce confusion between those two `convert()` method.

Reply:
@jdneo

I've renamed the `convert` method to `getExtension` and added a description.

I've also made it even simpler to get a language extension as you can just pass the sourceset...
```ScalaExtension scalaExtension = SupportedLanguages.SCALA.getExtension(sourceSet);```

I've also introduced `getLanguage` to the `LanguageModelBuilder` to prevent builders requesting the wrong `getLanguageCompileTask`

Reply:
New changes look good! Thank you!